src/NBitcoin.Tests/ChainTests.cs

```diff
@@ -170,21 +170,76 @@ public void CanBuildChain()
         [Trait("UnitTest", "UnitTest")]
         public void CanFindFork()
         {
-            var chain = new ConcurrentChain(this.network);
-            var chain2 = new ConcurrentChain(this.network);
-
-            this.AppendBlock(chain);
-            ChainedHeader fork = this.AppendBlock(chain);
-            ChainedHeader tip = this.AppendBlock(chain);
-
-            this.AssertFork(chain, chain2, chain.Genesis);
-            chain2 = new ConcurrentChain(this.networkTest);
-            this.AssertFork(chain, chain2, null);
-            chain2 = new ConcurrentChain(this.network);
-            chain2.SetTip(fork);
-            this.AssertFork(chain, chain2, fork);
-            chain2.SetTip(tip);
-            this.AssertFork(chain, chain2, tip);
+            var chain1 = new ConcurrentChain(this.network);
+
+				//create the main chain with a (commonChainSize) blocks before the fork
+				int commonChainSize = 150000;
```

Comment:
Is there benefit to adding this quantity of blocks? Would we get equivalent test coverage by decreasing it to something much lower?

Reply:
I let it as a parameter because i'm not sure, I added so much block because i thought there was a bug in the ChainHeader.FindFork and tried to see if it was happening with a longer chain.

With a very short chain, the skip list used by ConcurrentChain doesn't come into action at full, so this code was the reason i thought that with an higher chain, a bug could be exposed (but it didn't)
```
while (highChain.HashBlock != lowChain.HashBlock)
            {
                if (highChain.Skip != lowChain.Skip)
                {
                    highChain = highChain.Skip;
                    lowChain = lowChain.Skip;
                }
                else
                {
                    lowChain = lowChain.Previous;
                    highChain = highChain.Previous;
                }

                if ((lowChain == null) || (highChain == null))
                    return null;
            }
```

it takes 3s on my pc to test this method, anyway we can lower that parameter if it's a problem